File: xxf_arch/src/main/java/com/xxf/arch/core/activityresult/ActivityResult.java
package com.xxf.arch.core.activityresult;

import android.app.Activity;
import android.content.Intent;


public class ActivityResult {
    private int requestCode;
    private int resultCode;
    private Intent data;

    public ActivityResult(int requestCode, int resultCode, Intent data) {
        this.requestCode = requestCode;
        this.resultCode = resultCode;
        this.data = data;
    }

    public int getRequestCode() {
        return requestCode;
    }

    public int getResultCode() {
        return resultCode;
    }

    public Intent getData() {
        return data;
    }

    public boolean isOk() {
        return resultCode == Activity.RESULT_OK;
    }

    public boolean isCanceled() {
        return resultCode == Activity.RESULT_CANCELED;
    }
}

File: lib_activityresult/src/main/java/com/xxf/activityresult/ActivityResult.kt
package com.xxf.activityresult

import android.app.Activity
import android.content.Intent
import android.os.Parcel
import android.os.Parcelable

class ActivityResult(val requestCode: Int, val resultCode: Int, val data: Intent?) : Parcelable {
    val isOk: Boolean
        get() = resultCode == Activity.RESULT_OK
    val isCanceled: Boolean
        get() = resultCode == Activity.RESULT_CANCELED

    constructor(parcel: Parcel) : this(
        parcel.readInt(),
        parcel.readInt(),
        parcel.readParcelable(Intent::class.java.classLoader)
    ) {
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeInt(requestCode)
        parcel.writeInt(resultCode)
        parcel.writeParcelable(data, flags)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<ActivityResult> {
        override fun createFromParcel(parcel: Parcel): ActivityResult {
            return ActivityResult(parcel)
        }

        override fun newArray(size: Int): Array<ActivityResult?> {
            return arrayOfNulls(size)
        }
    }
}